common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationRequest.java

```diff
@@ -22,34 +22,258 @@
+     */
+    public void setAuthorizationEndpoint(String authorizationEndpoint) {
+        mAuthorizationEndpoint = authorizationEndpoint;
+    }
+
+    public void setPKCEChallenge(final PKCEChallenge pKCEChallenge) {
+        mPKCEChallenge = pKCEChallenge;
+    }
+
+    public String getResource() {
+        return mResource;
+    }
+
+    public void setResource(final String resource) {
+        mResource = resource;
+    }
+
+    public String getLoginHint() {
+        return mLoginHint;
+    }
+
+    public void setLoginHint(final String loginHint) {
+        mLoginHint = loginHint;
+    }
+
+    public UUID getCorrelationId() {
+        return mCorrelationId;
+    }
+
+    public void setCorrelationId(final UUID correlationId) {
+        mCorrelationId = correlationId;
+    }
+
+    public AADPromptBehavior getPromptBehavior() {
+        return mPromptBehavior;
+    }
+
+    public void setPromptBehavior(final AADPromptBehavior promptBehavior) {
+        mPromptBehavior = promptBehavior;
+    }
+
+    public String getExtraQP() {
+        return mExtraQP;
+    }
+
+    public void setExtraQP(final String extraQP) {
+        this.mExtraQP = extraQP;
+    }
+
+    public String getClaimsChallenge() {
+        return mClaimsChallenge;
+    }
+
+    public void setClaimsChallenge(final String claimsChallenge) {
+        mClaimsChallenge = claimsChallenge;
+    }
+
+
     public URL getAuthority() {
         return mAuthority;
     }
 
+    public void setAuthority(final URL authority) {
+        mAuthority = authority;
+    }
+
+
     /**
-     * @param authority URL
+     * @return String of ADAL version name.
      */
-    public void setAuthority(URL authority) {
-        mAuthority = authority;
+    public String getLibraryVersion() {
+        return mLibraryVersion;
+    }
+
+    /**
+     * The ADAL library should pass in its version.
+     *
+     * @param libraryVersion String of the library version name.
+     */
+    public void setLibraryVersion(final String libraryVersion) {
+        mLibraryVersion = libraryVersion;
     }
 
     //CHECKSTYLE:OFF
     @Override
     public String toString() {
         return "AzureActiveDirectoryAuthorizationRequest{" +
-                "mAuthority=" + mAuthority +
+                "mAuthority=" + mAuthorizationEndpoint +
                 "} " + super.toString();
     }
     //CHECKSTYLE:ON
+
+    public String getAuthorizationStartUrl() throws UnsupportedEncodingException, ClientException {
+        final String authorizationUrl = StringExtensions.appendQueryParameterToUrl(
+                mAuthorizationEndpoint,
+                createAuthorizationRequestParameters());
+        Logger.infoPII(TAG, mCorrelationId.toString(), "Request uri to authorize endpoint is: " + authorizationUrl);
+        return authorizationUrl;
+
+    }
+
+    private Map<String, String> createAuthorizationRequestParameters() throws UnsupportedEncodingException, ClientException {
```

Comment:
Can we shorten this method up? Break it into a few discrete functions?

Reply:
They can be tiny -- I really don't mind 1-line functions if they're descriptive. Methods that are too short don't bother me; ones that are too long lose me 😕